xmrstak/backend/cpu/minethd.cpp

```diff
@@ -475,17 +485,27 @@ void minethd::double_work_main()
 			continue;
 		}
 
+		size_t nonce_ctr = 250;
 		if(oWork.bNiceHash)
-			iNonce = calc_nicehash_nonce(*piNonce0, oWork.iResumeCnt);
+			iNonce = globalStates::inst().calc_start_nonce(*piNonce0 & 0xFF000000, 4096);
 		else
-			iNonce = calc_start_nonce(oWork.iResumeCnt);
+			iNonce = globalStates::inst().calc_start_nonce(0, 4096);
 
 		assert(sizeof(job_result::sJobID) == sizeof(pool_job::sJobID));
 
-		while (globalStates::inst().inst().iGlobalJobNo.load(std::memory_order_relaxed) == iJobNo)
+		while (globalStates::inst().iGlobalJobNo.load(std::memory_order_relaxed) == iJobNo)
 		{
 			if ((iCount & 0x7) == 0) //Store stats every 16 hashes
 			{
+				if(--nonce_ctr == 0)
```

Comment:
Can we separate the nonce_ctr from the statistic if. It is hard to read if the nonce counter is coupled with the statistic counter. This will add a if more but should not change the performance.

Reply:
@psychocrypt That's the reason why i added it there and not in the main loop. keep in mind that unlike the GPU miners, the loop is going fairly fast there.